### DISCUSSION TITLE
useIsMutating and useMutationState always trigger rerenders

### DISCUSSION THREAD

**Discussion by mleister97** [TanStack/query#6967] [Q&A]
Currently, the issue is that the two hooks, `useIsMutating` and `useMutationState`, always trigger a rerender, regardless of the specified `mutationKey` to filter. For instance, if I have a grid of 100x100, and each cell can have its own loading state, I would use a `mutationKey` like `['CELL', 3, 7]` for mutations. Unfortunately, all cells now trigger a rerender. Do you have any solution for this?

You can also see this when justing using a key for the filter which does not even exist, e.g.

```
// this does still rerender, regardless of the filter mutationKey
const isMutatingPosts = useIsMutating({ mutationKey: ['MUTATION_KEY_DOES_NOT_EVEN_EXIST'] })
```

**Comment by TkDodo** [comment]
can you show a reproduction please because that should not be the case

**Comment by mleister97** [comment]
Disregard that; I am unable to replicate the issue.

I will provide my 'reproducible' case as evidence that contradicts my previous statement: https://stackblitz.com/edit/tanstack-query-vba6tk?file=src%2Findex.jsx
